### PR TITLE
feat: init ssr-opengraph

### PR DESCRIPTION
**Pull request by preschian**
related https://github.com/kodadot/nft-gallery/issues/5268

the method is quite similar
1. if the request is coming from a non-bot https://github.com/kodadot/workers/pull/41/files#diff-1860af2eaf6e2c14cc185436d808d762199ab1e9ec51fd94afe805d52bf8570eR18 it will continue the request
2. if the request is coming from a crawler (bot) it will fetch data to subsquid and then render static html https://github.com/kodadot/workers/pull/41/files#diff-1860af2eaf6e2c14cc185436d808d762199ab1e9ec51fd94afe805d52bf8570eR54

todo:
- [ ] collection detail
- [ ] rmrk2
- [ ] put `rmrk` to `@kodadot1/static`?

step to deploy:
1. after merging this PR, make sure to deploy the `prerender-opengraph` worker first
2. and then deploy `ssr-opengraph`. to avoid routes conflict https://github.com/kodadot/workers/pull/41/files#diff-50d1a6082b03f808cc776ab435488ebf63e81e82829e73e9fc76d094f2d5199aR4

additional notes:
- seems like prerender on netlify not working anymore. so, it will works on `kodadot.xyz` only

### CODE REVIEW

**Review by vikiival — CHANGES_REQUESTED** [review]
please use `uniquery` and `minipfs`

**Comment by vikiival** [comment]
for the uniquery 

https://github.com/poppyseedDev/polkadot-fandom-page/blob/8274132d79f42967e992710951b27bd61dd111ac/src/pages/products/%5Bproduct%5D.tsx#LL26C9-L26C82

File: ssr-opengraph/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+export type Chain = 'rmrk' & Prefix;
+
+export const endpoints: Record<Chain, string> = {
+  bsx: 'https://squid.subsquid.io/snekk/v/005/graphql',
+  snek: 'https://squid.subsquid.io/snekk/v/004/graphql',
+  rmrk: 'https://squid.subsquid.io/rubick/graphql',
+};
+
+export const getNftById = async (chain: Chain, id: string) => {
+  const client = getClient();
```
Comment (vikiival) [inline ssr-opengraph/src/utils.ts:18]: If you want to use fetch on client you need pass prefix here 😇

**Review by vikiival — CHANGES_REQUESTED** [review]
Really good job, small things

**Comment by yangwao** [comment]
Somewhre on gaming nft marketplaces I've seen they've been pushing different numbers on card from collections, can create on that issue once I spot it again, that would be amazing to have it, otherwise looking forward to deploying this to beta & production

File: ssr-opengraph/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import type { Prefix } from '@kodadot1/static';
+import type { NFT, NFTMeta } from './types';
+
+// TODO: put 'rmrk' into Prefix type
+export type Chain = 'rmrk' & Prefix;
+
+export const endpoints: Record<Chain, string> = {
+  bsx: INDEXERS['bsx'],
+  snek: INDEXERS['snek'],
+  rmrk: 'https://squid.subsquid.io/rubick/graphql',
```
Comment (vikiival) [inline ssr-opengraph/src/utils.ts:15]:
```suggestion
  rmrk: INDEXERS['ksm'],
```

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
cc @preschian 

➜  prerender-opengraph git:(main) ✗ npx wrangler publish
 ⛅️ wrangler 2.1.13 (update available 2.13.0)
-------------------------------------------------------
Total Upload: 2.34 KiB / gzip: 1.02 KiB
Uploaded prerender-opengraph (3.64 sec)
Published prerender-opengraph (15.99 sec)
  kodadot.xyz/rmrk/detail/*
  kodadot.xyz/bsx/detail/*
  kodadot.xyz/snek/detail/*
  kodadot.xyz/glmr/gallery/*
  kodadot.xyz/glmr/detail/*
➜  prerender-opengraph git:(main) ✗ cd ../ssr-opengraph
➜  ssr-opengraph git:(main) ✗ npx wrangler publish
 ⛅️ wrangler 2.13.0
--------------------

✘ [ERROR] Missing entry-point: The entry-point should be specified via the command line (e.g. `wrangler publish path/to/script`) or the `main` config field.


If you think this is a bug then please create an issue at https://github.com/cloudflare/workers-sdk/issues/new/choose

**Comment by vikiival** [comment]
fixed in 6ee53cabff818bcaf720e8738f30c188b1d7bddf

**Pull request merged**